src/amg/azext_amg/tests/latest/test_amg_migrate_scenario.py

```diff
@@ -0,0 +1,335 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import tempfile
+import time
+import unittest
+
+
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer, MSGraphNameReplacer, MOCKED_USER_NAME)
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse
+
+from .test_definitions import (test_data_source, test_notification_channel, test_dashboard, test_dashboard_with_datasource)
+from .recording_processors import ApiKeyServiceAccountTokenReplacer
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+class AmgMigrateScenarioTest(ScenarioTest):
+
+    def __init__(self, method_name):
+        super().__init__(method_name, recording_processors=[
+            ApiKeyServiceAccountTokenReplacer()
+        ])
+
+    def _setup_migrate_instances(self):
+        # migrate from amg1 to amg2
+        amg1 = self.cmd('grafana create -g {rg} -n {name} -l {location}').get_output_in_json()
+        amg2 = self.cmd('grafana create -g {rg} -n {name2} -l {location}').get_output_in_json()
```

Comment:
migrating between two AMG instances for the test?

Reply:
Yep, it is easier to create two AMG instances. One of the instances is basically treated as a local so we get the service account token and that flow. If I were to try to do a local to AMG test, I will need to mock a lot more functions for basically the same effect.